### PR TITLE
removed week check from parts

### PR DESCRIPTION
**Pull request by abhisheksarka**
### Summary

Referencing this issue https://github.com/rails/rails/issues/34655, removed the week check

### CODE REVIEW

**Review by gmcgibbon — COMMENTED** [review]
I don't know a lot about date formatting, but according to [wikipedia](https://en.wikipedia.org/wiki/ISO_8601#Durations), this validation is necessary as mixing weeks and other duration units doesn't seem to be valid.

**Comment by rafaelfranca** [comment]
Yes. If anything it is `iso8601` method that is generating invalid value. It should not have `W` mixed with other things.

**Comment by abhisheksarka** [comment]
Yes, had similar thoughts on this but wasn't sure on the approach. I will take this up and make the changes

**Comment by abhisheksarka** [comment]
I looked at other libs as to how they are handling this use case. Specifically `moment`, converts weeks to days which is the change I pushed to get around this issue.
 
Sample Output

```ruby
duration = ActiveSupport::Duration.build(1000000)
# 1 week, 4 days, 13 hours, 46 minutes, and 40.0 seconds

duration_iso = duration.iso8601
# P11DT13H46M40S

ActiveSupport::Duration.parse(duration_iso)
# 11 days, 13 hours, 46 minutes, and 40 seconds
```

WDYT?

**Comment by abhisheksarka** [comment]
Anyone?

File: activesupport/test/core_ext/duration_test.rb [hunk 1]
```diff
@@ -559,7 +559,7 @@ def test_iso8601_parsing_wrong_patterns_with_raise
   def test_iso8601_output
     expectations = [
       ["P1Y",           1.year                           ],
-      ["P1W",           1.week                           ],
+      ["P7D",           1.week                           ],
```
Comment (gmcgibbon) [inline activesupport/test/core_ext/duration_test.rb]: I think we should still expect `P1W` if there are only weeks in the duration, right? Try adding another test that better captures this edge case with weeks and other parts.

**Review by gmcgibbon — COMMENTED** [review]
This seems like an acceptable approach to me, but I think we should only convert weeks to days if there are multiple duration parts. Also, please add a changelog entry.

**Comment by abhisheksarka** [comment]
shall we proceed with this fix?

**Comment by abhisheksarka** [comment]
@gmcgibbon ??

File: activesupport/lib/active_support/duration/iso8601_serializer.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@ module ActiveSupport
   class Duration
     # Serializes duration to string according to ISO 8601 Duration format.
     class ISO8601Serializer # :nodoc:
+      DATE_COMPONENTS = [:years, :months, :days]
```
Comment (gmcgibbon) [inline activesupport/lib/active_support/duration/iso8601_serializer.rb]: Nitpick: you can use `%i()` notation here.

File: activesupport/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,29 @@
...
+    ```ruby
+    duration = ActiveSupport::Duration.build(1000000)
+    # 1 week, 4 days, 13 hours, 46 minutes, and 40.0 seconds
+
+    duration_iso = duration.iso8601
+    # P11DT13H46M40S
+
+    ActiveSupport::Duration.parse(duration_iso)
+    # 11 days, 13 hours, 46 minutes, and 40 seconds
+
```
Comment (gmcgibbon) [inline activesupport/CHANGELOG.md]: ✂️ I think you can remove this extra line.

File: activesupport/lib/active_support/duration/iso8601_serializer.rb [hunk 4]
```diff
@@ -19,8 +21,8 @@ def serialize
         output = +"P"
         output << "#{parts[:years]}Y"   if parts.key?(:years)
         output << "#{parts[:months]}M"  if parts.key?(:months)
-        output << "#{parts[:weeks]}W"   if parts.key?(:weeks)
         output << "#{parts[:days]}D"    if parts.key?(:days)
+        output << "#{parts[:weeks]}W"   if parts.key?(:weeks) && !week_mixed_with_date?(parts)
```
Comment (gmcgibbon) [inline activesupport/lib/active_support/duration/iso8601_serializer.rb]: Why do we need `&& !week_mixed_with_date?(parts)` here if normalize is always ran before this, removing weeks?

**Review by gmcgibbon — COMMENTED** [review]
Can you squash your commits into one?

**Comment by kamipo** [comment]
Merged in #38046.

**Pull request closed without merging**